src/Query.jl

```diff
@@ -0,0 +1,243 @@
+module Query
+
+export From, Where, Select, AbstractQueryFormatter, SimpleQueryFormatter, NamedQueryFormatter, DFQueryFormatter
+
+using ..ACSetInterface, ..Schemas
+using MLStyle
+using DataFrames: DataFrame
+using StructEquality
+
+to_name(x) = x
+to_name(x::Pair) = Symbol("$(x.second)$(x.first)")
+to_name(x::Val{T}) where T = Symbol("Val_$T")
+
+""" This function lifts a singleton into an iterator. This prevents branches in the control flow.
+```
+a=1; @test [k for k ∈ iterable(a)] == [1]
+a=[1,2,3]; @test [k for k ∈ iterable(a)] == [1,2,3]
+```
+"""
+function iterable(x::T) where T
+  S = T <: AbstractVector ? eltype(T) : T
+  [S[]; x]
+end
+
+function Base.get(acset::ACSet, select::Symbol, idx=Colon(); schema=acset_schema(acset))
+  val = select ∈ objects(schema) ? parts(acset, select) : subpart(acset, select)
+  val[idx]
+end
+
+function Base.get(acset, selects::Vector{Symbol}, idx=Colon(); kwargs...)
+  zip(get.(Ref(acset), selects, idx; kwargs...)...)
+end
+
+abstract type AbstractCondition end
```

Comment:
Helpful to have some small docstring here, even if it's just "AbstractCondition is a common type for WhereCondition, AndWhere, and OrWhere"